Makefile

```diff
@@ -19,7 +20,7 @@ commands :
 
 ## docker-serve     : use docker to build the site
 docker-serve :
-	docker run --rm -it -v ${PWD}:/srv/jekyll -p 127.0.0.1:4000:4000 jekyll/jekyll:${JEKYLL_VERSION} make serve
+	docker run --rm -it -v ${PWD}:/srv/jekyll -p 4000:4000 jekyll/jekyll:${JEKYLL_VERSION} ${JEKYLL_DOCKER}
```

Comment:
:wave: @fmichonneau ! This looks like a great start but I had trouble running this on my system, I kept getting these errors:

```
Bundle updated!
/bin/sh: 1: bundle: not found
Makefile:23: recipe for target 'docker-serve' failed
make: *** [docker-serve] Error 127
```

A few other things to note:

1. we may still want to leave the 127.0.0.1 in the docker port mapping so we aren't exposing the jekyll server to the network
2. I think we should install the bundle to `./vendor/bundle` (mapped onto the local filesystem) so we don't have to download/install/update the gems every single time we run the application. 
3. given the increasing complexity of the docker command, it might be good to have a standalone shell script (still very simple) like `jekyll.sh` that does the bundle install, bundle update, and bundle exec jekyll serve.

Reply:
Thanks for your work and feedback on this!

Were you getting the error your report using the `latest` jekyll image?

I'll respond to your notes with some comments:

1. In my tests, and from my reading, if we use 127.0.0.1, it doesn't work on Ubuntu by default. Which OS do you use? Do you know if there are settings elsewhere on the system or Docker that need to be changed to make it work? The networking part of Docker isn't something I'm familiar with.

2. I have tried to this, but in my tests, the file permissions interacted in a way that made it impossible to use both the Docker-based approach and the local version of bundler/jekyll. I can imagine that most people who would use Docker will not use the local executables and that might be OK. It'd be great to have more feedback from maintainers who use the Docker-based approach to understand better how they use it. There might also be a way to work around this that I don't know about!

3. That sounds good. We may want to put this script in the `bin/` folder.

Reply:
I got the error using jekyll 3.8.5 as specified in the Makefile - still not exactly sure why it is happening but I think it has to do with the `&&` chaining which doesn't work as we want it to when run via `docker exec`. I vaguely recall having to run chained commands via `sh -c "foo && bar && baz"` or some other hacky workaround, for which point 3 is a better solution (having a dedicated shell script).

I've got a working example of this at https://github.com/alee/styles/tree/use-bundler and there's a working version of this applied to the python-novice-gapminder lesson at https://github.com/alee/python-novice-gapminder/tree/docker-bundle

You can see the commit where all these changes were made at https://github.com/alee/styles/commit/00245b1bb83e707c2db5bd7a0af16423db693707

1. The docker container definitely needs the --host 0.0.0.0 but here we are binding the docker container's exposed port 4000 to our localhost:4000 so that when we visit localhost:4000 on our host machine, it'll proxy to the docker container's port 4000. I think the issue you may have run into was the missing `jekyll serve --host 0.0.0.0` which I'm guessing is implicit in the normal `jekyll serve` configuration but not the `bundle exec jekyll serve` for some reason..

2. In the code I linked above I run the `bundle install --path .vendor/bundle` so after running `make docker-serve` you should see a new docker managed & created `.vendor/` directory in the root of your repository (which is why there's an additional .gitignore entry there as well). I am not sure where people tend to put there typical bundle install libs but we can pick a different hidden directory as well (like `.docker-vendor/bundle`) that should provide a fairly strong guarantee of not conflicting with their system install

3. Good call!

Let me know if you'd like me to drop those commits into this PR, though some permission adjustment is probably needed to allow me to do that